src/model/internal.rs

```diff
@@ -0,0 +1,105 @@
+//! Contains the internal storage definition of a manifest
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+use crate::model::Manifest;
+use crate::storage::StateKind;
+
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub(crate) struct StoredManifest {
```

Comment:
Potential nit: this struct represents a single manifest with all of its version history, but that wasn't immediately clear. I thought it might be storage of all current manifests until I kept reading. A comment or different type for the key into the index map would help